pulldown-cmark/src/parse.rs

```diff
@@ -617,14 +618,97 @@ impl<'input, F: BrokenLinkCallback<'input>> Parser<'input, F> {
+                                        // the display text
+                                        self.tree[body_node].item.start = start_ix + rest;
+                                        Some((body_node, wikiname))
+                                    } else {
+                                        None
+                                    }
+                                }
+                                None => {
+                                    // [[WikiName]]
+                                    let wikiname = &block_text[start_ix..end_ix];
+                                    // or [[Nested/WikiName]]
+                                    let display_ix = wikiname
+                                        .as_bytes()
+                                        .iter()
+                                        .rposition(|b| *b == b'/')
+                                        .map(|ix| ix + 1)
+                                        .unwrap_or(0)
+                                        + start_ix;
+                                    // TODO: wikitext should not be styled, might
+                                    // need a more experienced contributor's help
+                                    let body_node = self.tree.create_node(Item {
+                                        start: display_ix,
+                                        end: end_ix,
+                                        body: ItemBody::Text {
+                                            backslash_escaped: false,
+                                        },
+                                    });
+                                    Some((body_node, wikiname))
+                                }
+                            };
+                            // exists only to panic guard against edge cases, this
+                            // should run 99.9% of the time
+                            if let Some((body_node, wikiname)) = wikilink {
+                                let link_ix = self.allocs.allocate_link(
+                                    LinkType::WikiLink,
+                                    format_wikilink(wikiname),
+                                    "".into(),
+                                    "".into(),
+                                );
+                                self.tree[outer_el.node].item.body = ItemBody::Link(link_ix);
+                                self.tree[outer_el.node].child = Some(body_node);
+                                self.tree[outer_el.node].next = next_node;
+                                self.tree[outer_el.node].item.end = end_ix + 1;
+                                self.link_stack.disable_all_links();
+                            }
+                            // at this point, regardless of whether we were
+                            // successful the stack has been trashed, so use this
+                            // to bring the pass back in a valid state
```

Comment:
Ordinary markdown links are allowed to start or end with `[[` and `]]`, without the other side also being doubled up (meaning it's not a wikilink).

For example:

```markdown
[[squid] calamari is considered a delicacy](https://en.wikipedia.org/wiki/Squid)

[calamari [squid]](https://en.wikipedia.org/wiki/Squid)
```

> [[squid] calamari is considered a delicacy](https://en.wikipedia.org/wiki/Squid)
>
> [calamari [squid]](https://en.wikipedia.org/wiki/Squid)

This works in commonmark-hs with wikilinks turned on ([playground](https://pandoc.org/try/?params={%22text%22%3A%22[[squid]+calamari+is+considered+a+delicacy](https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FSquid)\n\n[calamari+[squid]](https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FSquid)%22%2C%22to%22%3A%22html5%22%2C%22from%22%3A%22commonmark%2Bwikilinks_title_after_pipe%22%2C%22standalone%22%3Afalse%2C%22embed-resources%22%3Afalse%2C%22table-of-contents%22%3Afalse%2C%22number-sections%22%3Afalse%2C%22citeproc%22%3Afalse%2C%22html-math-method%22%3A%22plain%22%2C%22wrap%22%3A%22auto%22%2C%22highlight-style%22%3Anull%2C%22files%22%3A{}%2C%22template%22%3Anull})), and should work in any implementation that parses wikilinks in a separate pass (assuming they correctly "bail out" if none are found), such as Gollum.

It doesn't work correctly in this PR, when wikilinks are turned on.